docs/DEVICE-CONNECTIVITY.md

```diff
@@ -1,142 +1,483 @@
+Load spreading means that there are two or more similar uplink networks, for instance
+two Ethernet ports for redundancy, the EVE will send different requests over different
+connections in a round-robin fashion.
+
+Failover means that a device can have different types of uplink networks, for example
+Ethernet, LTE, and/or satellite connectivity, and switch from one to another to maintain
+controller connectivity. In such case, a cost (field ```cost``` of the
+[SystemAdapter message](https://github.com/lf-edge/eve-api/tree/main/proto/config/devmodel.proto))
+can be assigned to each uplink port so that e.g., LTE (the wwan0 port) is only used if EVE
+can not connect via any lower cost (e.g., one or more Ethernet) ports.
+
+The cost is a number between 0 and 255, with zero (the default) implying free,
+and less preferred ports being assigned a higher cost. Multiple ports can be assigned
+the same cost, in which case once failover has happened to cost N, then all uplink ports
+with cost N will be used for load spreading of the management traffic.
+
+Furthermore, one can set [network.download.max.cost](CONFIG-PROPERTIES.md) to a number N
+if it is acceptable for EVE to perform (potentially large) downloads of content and images
+when having failed over to uplink ports with cost N (or less).
+
+The cost is new and will replace the (now obsolete) `free`/`freeUplink` way to specify
+two levels of cost (free or paid). For compatibility reasons, EVE will look at `freeUplink`
+for the `SystemAdapter` and `PhysicalIO` so that the free/paid distinction still works
+until the cost parameter is used by all the controllers.
+
+## Network Proxies
+
+Organizations frequently utilize network proxies to regulate and supervise internet access,
+apply content filtering, and enforce security policies.
+Depending on the proxy type, adjustments to the edge-node network configuration may be necessary
+to ensure connectivity with the cloud.
+
+First of all, before configuring a device, the user must find out if there is a network proxy
+in the network and understand how it is configured. Configuration will then depend on the type
+of the proxy, see below.
+
+### Transparent network proxy
+
+A network proxy may operate *transparently*, meaning that clients (EVE devices, user apps)
+making HTTP/HTTPS/… requests are not aware of its presence. When a client makes an HTTP(S)
+request, it addresses it directly to the destination (e.g. EVE controller), but a transparent
+proxy will capture it (without the client knowing about it) and apply a proxying action.
+The action may depend on the destination URL. For example, the proxy may be configured to drop
+requests heading to some specific domains, but allow others.
+
+Additionally, the proxy may break the TLS tunnel into two (client-proxy, proxy-destination)
+and act as a MITM, reading the otherwise encrypted traffic. This is done by proxy generating
```

Comment:
I don't know if we already spell out MiTM - if not it makes sense to do e.g. "Man-in-the-Middle (MITM)" the first time we use it.

Reply:
Done